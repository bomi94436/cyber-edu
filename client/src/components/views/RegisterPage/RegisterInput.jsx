import React from "react";

const RegisterRow = ({
  title,
  type,
  placeholder,
  dispatchName,
  setRegister,
}) => {
  return (
    <div>
      <label htmlFor={dispatchName + "-input"}>{title}</label>
      <input
        id={dispatchName + "-input"}
        type={type}
        placeholder={placeholder}
        onChange={(event) =>
          setRegister({ name: dispatchName, value: event.target.value })
        }
      />
      <br />
    </div>
  );
};

export default RegisterRow;
