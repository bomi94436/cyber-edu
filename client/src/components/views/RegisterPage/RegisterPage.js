import React from "react";
import { connect } from "react-redux";
import { setRegister, userRegister } from "../../../modules/main";

const RegisterPage = ({ setRegister, userRegister }) => {
  const handleSubmit = (e) => {
    e.preventDefault();
    userRegister();
  };
  return (
    <div>
      <form onSubmit={(e) => handleSubmit(e)}>
        <span>학번</span>
        <input
          type="text"
          placeholder="학번"
          onChange={(event) =>
            setRegister({ name: "studentId", value: event.target.value })
          }
        />
        <br />
        <span>비밀번호</span>
        <input
          type="password"
          placeholder="비밀번호"
          onChange={(event) =>
            setRegister({ name: "password", value: event.target.value })
          }
        />
        <br />
        <button type="submit">회원가입</button>
      </form>
    </div>
  );
};

export default connect(
  (state) => ({}),
  (dispatch) => ({
    setRegister: (data) => dispatch(setRegister(data)),
    userRegister: () => dispatch(userRegister()),
  })
)(RegisterPage);
