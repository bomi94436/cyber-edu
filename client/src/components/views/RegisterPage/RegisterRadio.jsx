import React from "react";

const RegisterRadio = ({ title, name, value, checked, setRegister }) => {
  return (
    <span>
      <label htmlFor={name + "-input"}>{title}</label>
      <input
        id={name + "-input"}
        type="radio"
        name={name}
        value={value}
        defaultChecked={checked}
        onChange={(event) =>
          setRegister({ name: name, value: event.target.value })
        }
      />
    </span>
  );
};

export default RegisterRadio;
