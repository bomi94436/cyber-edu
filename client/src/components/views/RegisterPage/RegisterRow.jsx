import React from "react";
import { connect } from "react-redux";
import { setRegister } from "../../../modules/main";

const RegisterRow = ({
  title,
  type,
  placeholder,
  dispatchName,
  setRegister,
}) => {
  return (
    <div>
      <span>{title}</span>
      <input
        type={type}
        placeholder={placeholder}
        onChange={(event) =>
          setRegister({ name: dispatchName, value: event.target.value })
        }
      />
      <br />
    </div>
  );
};

export default connect(
  (state) => ({}),
  (dispatch) => ({
    setRegister: (data) => dispatch(setRegister(data)),
  })
)(RegisterRow);
