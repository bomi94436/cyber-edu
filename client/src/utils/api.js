import axios from "axios";
import { USER_SERVER } from "../components/Config";

export const login = (dataToSubmit) =>
  axios.post(`${USER_SERVER}/login`, dataToSubmit);

export const register = (dataToSubmit) =>
  axios.post(`${USER_SERVER}/register`, dataToSubmit);
