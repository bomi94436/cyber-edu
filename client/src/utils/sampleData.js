export const register = [
  { title: "학생", name: "student", value: "professor" },
  { title: "학번(직번)", name: "studentId", value: "201795076" },
  { title: "비밀번호", name: "password", value: "bomibomi" },
  { title: "이름", name: "name", value: "정보미" },
  { title: "이메일", name: "email", value: "boo0729611@gmail.com" },
  { title: "휴대폰 번호", name: "phone", value: "01012345678" },
];
