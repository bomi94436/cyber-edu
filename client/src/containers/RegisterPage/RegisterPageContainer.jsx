import RegisterPage from "../../components/views/RegisterPage/RegisterPage";
import { connect } from "react-redux";
import { setRegister, postRegister } from "../../modules/main";

const RegisterPageContainer = connect(
  (state) => ({
    state: state.register,
  }),
  (dispatch) => ({
    setRegister: (data) => dispatch(setRegister(data)),
    postRegister: (data) => dispatch(postRegister(data)),
  })
)(RegisterPage);

export default RegisterPageContainer;
