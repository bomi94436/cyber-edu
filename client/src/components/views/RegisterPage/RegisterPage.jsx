import React from "react";
import RegisterInput from "./RegisterInput";
import RegisterRadio from "./RegisterRadio";

const RegisterPage = ({ stateReigster, setRegister, postRegister }) => {
  const handleSubmit = (e) => {
    e.preventDefault();
    postRegister(stateReigster);
  };
  return (
    <div>
      <form onSubmit={(e) => handleSubmit(e)}>
        <RegisterRadio
          title="학생"
          name="role"
          value="student"
          checked
          setRegister={setRegister}
        />
        <RegisterRadio
          title="교수"
          name="role"
          value="professor"
          setRegister={setRegister}
        />
        <RegisterInput
          title="학번(직번)"
          type="text"
          placeholder="202012345 ..."
          dispatchName="studentId"
          setRegister={setRegister}
        />
        <RegisterInput
          title="비밀번호"
          type="password"
          placeholder="password ..."
          dispatchName="password"
          setRegister={setRegister}
        />
        <RegisterInput
          title="이름"
          type="text"
          placeholder="홍길동 ..."
          dispatchName="name"
          setRegister={setRegister}
        />
        <RegisterInput
          title="이메일"
          type="email"
          placeholder="test@example.com ..."
          dispatchName="email"
          setRegister={setRegister}
        />
        <RegisterInput
          title="휴대폰 번호"
          type="text"
          placeholder="01012345678 ..."
          dispatchName="phone"
          setRegister={setRegister}
        />
        <button type="submit">회원가입</button>
      </form>
    </div>
  );
};

export default RegisterPage;
