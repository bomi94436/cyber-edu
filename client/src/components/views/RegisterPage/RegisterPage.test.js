import React from "react";
import { render, fireEvent } from "@testing-library/react";
import RegisterPage from "./RegisterPage";
import * as data from "../../../utils/sampleData";

describe("<RegisterPage />", () => {
  const mockChange = jest.fn();
  const mockSubmit = jest.fn();

  const setup = () => {
    const component = render(
      <RegisterPage setRegister={mockChange} postRegister={mockSubmit} />
    );
    return component;
  };

  it("renders correctly", () => {
    const component = setup();
  });

  it("matches snapshot", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

  it("has labels, inputs and button", () => {
    const component = setup();

    data.register.forEach((element) => {
      component.getByText(element.title);
      component.getByLabelText(element.title);
    });
    component.getByText("회원가입");
  });

  it("calls onChange", () => {
    const component = setup();

    data.register.forEach((element) => {
      const input = component.getByLabelText(element.title);
      const value = element.value;
      const mockedEvent = {
        target: {
          value: element.value,
        },
      };
      fireEvent.change(input, mockedEvent);
      expect(input.value).toBe(value);
    });
  });

  it("calls onSubmit", () => {
    const component = setup();
    fireEvent.submit(component.getByText("회원가입"));
    expect(mockSubmit).toHaveBeenCalled();
  });
});
