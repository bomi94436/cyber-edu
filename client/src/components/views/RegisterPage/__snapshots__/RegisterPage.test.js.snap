// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <span>
            <label
              for="role-input"
            >
              학생
            </label>
            <input
              checked=""
              id="role-input"
              name="role"
              type="radio"
              value="student"
            />
          </span>
          <span>
            <label
              for="role-input"
            >
              교수
            </label>
            <input
              id="role-input"
              name="role"
              type="radio"
              value="professor"
            />
          </span>
          <div>
            <label
              for="studentId-input"
            >
              학번(직번)
            </label>
            <input
              id="studentId-input"
              placeholder="202012345 ..."
              type="text"
            />
            <br />
          </div>
          <div>
            <label
              for="password-input"
            >
              비밀번호
            </label>
            <input
              id="password-input"
              placeholder="password ..."
              type="password"
            />
            <br />
          </div>
          <div>
            <label
              for="name-input"
            >
              이름
            </label>
            <input
              id="name-input"
              placeholder="홍길동 ..."
              type="text"
            />
            <br />
          </div>
          <div>
            <label
              for="email-input"
            >
              이메일
            </label>
            <input
              id="email-input"
              placeholder="test@example.com ..."
              type="email"
            />
            <br />
          </div>
          <div>
            <label
              for="phone-input"
            >
              휴대폰 번호
            </label>
            <input
              id="phone-input"
              placeholder="01012345678 ..."
              type="text"
            />
            <br />
          </div>
          <button
            type="submit"
          >
            회원가입
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <span>
          <label
            for="role-input"
          >
            학생
          </label>
          <input
            checked=""
            id="role-input"
            name="role"
            type="radio"
            value="student"
          />
        </span>
        <span>
          <label
            for="role-input"
          >
            교수
          </label>
          <input
            id="role-input"
            name="role"
            type="radio"
            value="professor"
          />
        </span>
        <div>
          <label
            for="studentId-input"
          >
            학번(직번)
          </label>
          <input
            id="studentId-input"
            placeholder="202012345 ..."
            type="text"
          />
          <br />
        </div>
        <div>
          <label
            for="password-input"
          >
            비밀번호
          </label>
          <input
            id="password-input"
            placeholder="password ..."
            type="password"
          />
          <br />
        </div>
        <div>
          <label
            for="name-input"
          >
            이름
          </label>
          <input
            id="name-input"
            placeholder="홍길동 ..."
            type="text"
          />
          <br />
        </div>
        <div>
          <label
            for="email-input"
          >
            이메일
          </label>
          <input
            id="email-input"
            placeholder="test@example.com ..."
            type="email"
          />
          <br />
        </div>
        <div>
          <label
            for="phone-input"
          >
            휴대폰 번호
          </label>
          <input
            id="phone-input"
            placeholder="01012345678 ..."
            type="text"
          />
          <br />
        </div>
        <button
          type="submit"
        >
          회원가입
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
